src/api/hooks/useSubmitTransaction.ts

```diff
@@ -0,0 +1,36 @@
+import {TxnBuilderTypes} from "aptos";
+import {useState} from "react";
+import {signAndSubmitTransaction} from "../wallet";
+
+export type TransactionResponse =
+  | TransactionResponseOnSuccess
+  | TransactionResponseOnFailure;
+
+export type TransactionResponseOnSuccess = {
+  succeeded: boolean;
```

Comment:
Is it possible for succeeded to be false here? I think the type could be `succeeded: true`?